_posts/2020-04-01-Robust Apps for the cloud.md

```diff
@@ -0,0 +1,253 @@
+
+**Learn More**
+
+- [Set up staging environments in Azure App Service](https://docs.microsoft.com/azure/app-service/deploy-staging-slots)
+- [Azure Web App Deployment Slot Swap with Preview](https://ruslany.net/2015/10/azure-web-app-deployment-slot-swap-with-preview/)
+- [Deployment best practices](https://docs.microsoft.com/azure/app-service/deploy-best-practices#use-deployment-slots) 
+
+## Set your Health Check path
+
+App Service allows you to specify a health check path on your apps. The platform pings this path to determine if your application is healthy and responding to requests. When your site is scaled out to multiple instances, App Service will exclude any unhealthy instance(s) from serving requests, improving your overall availability. Your app's health check path should poll the critical components of your application, such as your database, cache, or messaging service. This ensures that the status returned by the health check path is an accurate picture of the overall health of your application.
+
+1. Go to *Development Tools* > *Resource Explorer* on the web app blade for Azure portal:
+
+    ![health-check-1]({{site.baseurl}}/media/2020/04/health-check-1.jpg)
+
+1. On the resource explorer page, expand the "config" section and click the "web" tab. Add an element with the name, "healthCheckPath", and value is the path of your health-check url that our service will ping.
+
+    ![health-check-2]({{site.baseurl}}/media/2020/04/health-check-2.png)
+
+> Please note that the Health Check feature works only when you have one or more instances, which is a **very** strong recommendation. For a single instance web app, the traffic is never blocked even if that single instance is encountering issues.
+
+**Learn More**
+
+- [Health Check (Preview)](https://github.com/projectkudu/kudu/wiki/Health-Check-(Preview))
+
+## Use Application Initialization
+
+Enable Application Initialization to ensure that your app instances are have fully started before they are added to the load balancer. Application Initialization is used during site restarts, auto scaling, and manual scaling. This is a critical feature where hitting the site's root path is not sufficient to start the application. For this purpose a warm-up path must be created on the app which should be unauthenticated and App Init should be configured to use this url path. Try to make sure that the method implemented by the warm-up url takes care of touching the functions of all important routes and it returns a response only when warm-up is complete. The site will be put into production only when it returns a response (success or failure) and app initialization will assume "everything is fine with the app". App Initialization can be configured for your app within web.config file.
```

Comment:
There is a lot of information thrown at the reader here. I think some of this can be cut and referred to in the docs. If you want all the info on the article, then this should probably be in two paragraphs and explain more.

Reply:
I reworded it a bit and split it into two paragraphs.